Add 2021 Community Survey Results Blog (#923)
rust-lang/blog.rust-lang.org: posts/2022-02-15-Rust-Survey-2021.md

English, however, is not the language of choice for all Rustaceans with nearly 7% preferring not to use English as a language for technical communication. An additional 23% of respondents prefer another language in addition to English. The most commonly preferred languages (besides English) roughly follow where Rustaceans live with Chinese, German, and French being the top 3. However, Japanese, Chinese, and Russian speakers were the most likely to prefer not to use English at all for technical conversation.

## Using Rust

The percentage of people using Rust continues to rise. Of those using Rust, 81% are currently using it on at least a weekly basis compared to 72% from last year's survey.

75% of all Rust users say they can write production ready code, though 27% say that it is at times a struggle.

![Rust expertise](/images/2022-02-RustSurvey/expertise.png)

Overall, Rustaceans seem to be having a great time writing Rust with only 1% saying it isn't fun to use. Only a quarter of a percent find Rust doesn't have any real benefit over other programming languages.

## Rust at Work

> Rust can now safely be classified as a language used by people in professional settings. 59% of respondents who currently use Rust, use it at least occasionally at work with 23% using Rust for the majority of their coding. This is a large increase over last year where 58% of respondents didn't use Rust at work compared with this year's 41%.

Reviewer: I had to read this paragraph several times. It particularly confused me that it switched from "percent that do use rust" to "percent that don't" when doing a comparison. Suggested rewrite.

```suggestion
Rust can now safely be classified as a language used by people in professional settings. Of those respondents using Rust, 59% use it at least occasionally at work with 23% using Rust for the majority of their coding. This is a large increase over last year where only 42% of respondents used Rust at work.
```

![Do you use Rust at work?](/images/2022-02-RustSurvey/rust-at-work.png)

> As for why respondents are using Rust at work, the top answer was that it allowed users "to build relatively correct and bug free software" with 96% of respondents agreeing with that statement. After correctness, performance (92%) was the next most popular choice. 89% of respondents agreed that they picked Rust at work because of Rust's much-discussed security properties. Perhaps surprisingly, 89% of respondents also chose Rust because their teams found it fun and enjoyable to program (though respondents did agree more strongly with their choice to pick Rust due to its security properties).
> 
> Adopting Rust at work seems to follow a challenging but ultimately worthwhile path for a lot of Rustaceans. First, 83% of those who have adopted Rust at work found it to be "challenging". How much this is related to Rust itself versus general challenges with adopting a new programming language, however, is unclear. During adoption only 13% of respondents found the language was slowing their team down; which contrasts with the 82% who found that Rust helped their teams achieve their goals.
> 
> After adoption, the costs seem to be worth it: only 1% of respondents did not find the challenge worth it while 79% said it definitely was. When asked if their teams were likely to use Rust again in the future, 90% agreed.

Reviewer 2: I wonder if we should re-arrange some of this to more explicitly state the progression path from challenging to helping achieve their goals to fun and enjoyable to program in. Something like:

```suggestion
Adopting Rust at work seems to follow a long but ultimately worthwhile path for a lot of Rustaceans. First, 83% of those who have adopted Rust at work found it to be "challenging". How much this is related to Rust itself versus general challenges with adopting a new programming language, however, is unclear. During adoption only 13% of respondents found the language was slowing their team down and 82% found that Rust helped their teams achieve their goals. 

After adoption, the costs seem to be justified: only 1% of respondents did not find the challenge worth it while 79% said it definitely was. When asked if their teams were likely to use Rust again in the future, 90% agreed. Finally, of respondents using Rust at work, 89% of respondents said their teams found it fun and enjoyable to program.

As for why respondents are using Rust at work, the top answer was that it allowed users "to build relatively correct and bug free software" with 96% of respondents agreeing with that statement. After correctness, performance (92%) was the next most popular choice. 89% of respondents agreed that they picked Rust at work because of Rust's much-discussed security properties. 
```

Author: Love it!

The IDE experience (led through continued adoption and improvement of various tools like rust-analyzer, IntelliJ Rust, etc.) gets the prize for showing the most improvement:  56% found it has improved over the last year.

However, compiler error messages received the most praise, with 90% approval of their current state. 🎉

When asked what their biggest worries for the future of Rust are, the top answer was a fear that there will not be enough usage in industry (38%). Given that Rust continues to show strides in adoption at places of work, the community seems to be on a good path to overcoming this concern.

The next largest concern was that the language would become too complex (33%). This was combined with a relative small number of folks calling for additional features (especially for ones not already in the pipeline).

> Finally, the third largest concern was that those working on Rust would not find the proper support they need to continue to develop the language and community in a healthy way. With the establishment of the Rust Foundation, support structures are coming into place that hopefully will address this point, but no doubt plenty of work is still ahead of us.

Reviewer 3: The previous two paragraphs had a percentage; could this one add a percentage as well?
